calendar-rest-api/src/main/java/com/linagora/calendar/restapi/routes/ResourceParticipationRoute.java

```diff
@@ -0,0 +1,196 @@
+                if (exception instanceof ResourceNotFoundException
+                    || exception instanceof CalendarEventNotFoundException) {
+                    return handleNotFound(response, exception);
+                }
+                return handleServerError(response, exception);
+            });
+    }
+
+    private Mono<Void> updateResourceParticipation(Resource resource,
+                                                   String eventPathId,
+                                                   PartStat partStat) {
+        return openPaaSDomainDAO.retrieve(resource.domain())
+            .flatMap(domain -> calDavEventRepository.updatePartStat(domain, resource.id(), eventPathId, partStat));
+    }
+
+    private Mono<Void> authorizeResourceAdmin(MailboxSession session,
+                                              List<ResourceAdministrator> administrators) {
+        if (administrators == null || administrators.isEmpty()) {
+            return Mono.error(new ForbiddenException("Resource has no administrators defined"));
+        }
+
+        return openPaaSUserDAO.retrieve(session.getUser())
+            .filter(openPaaSUser -> administrators.stream()
+                .anyMatch(admin -> admin.refId().equals(openPaaSUser.id())))
+            .switchIfEmpty(Mono.error(new ForbiddenException("User does not have admin permission for this resource")))
+            .then();
+    }
+
+    private ResourceId extractResourceId(HttpServerRequest request) {
+        return new ResourceId(request.param(RESOURCE_ID_PATH_PARAM));
+    }
+
+    private String extractEventPathId(HttpServerRequest request) {
+        return request.param(EVENT_PATH_ID_PATH_PARAM);
+    }
+
+    private PartStat validateAndExtractPartStatParam(HttpServerRequest request) {
+        return getParameter(request, PART_STAT_QUERY_PARAM)
+            .map(PartStat::new)
+            .filter(partStat -> PartStat.ACCEPTED.equals(partStat) || PartStat.DECLINED.equals(partStat))
+            .orElseThrow(() -> new IllegalArgumentException("Missing or invalid partStat parameter: " + PART_STAT_QUERY_PARAM));
+    }
+
+    private void validateReferrer(HttpServerRequest request) {
+        String referrer = getParameter(request, REFERRER_QUERY_PARAM)
+            .orElseThrow(() -> new IllegalArgumentException("Missing referrer parameter: " + REFERRER_QUERY_PARAM));
+        if (!EXPECTED_REFERRER.equalsIgnoreCase(referrer)) {
+            throw new IllegalArgumentException("Invalid referrer parameter, expected: " + EXPECTED_REFERRER);
+        }
+    }
+
+    private Optional<String> getParameter(HttpServerRequest request, String parameterName) {
+        return new QueryStringDecoder(request.uri()).parameters().getOrDefault(parameterName, List.of())
+            .stream()
+            .filter(StringUtils::isNotBlank)
+            .findAny();
+    }
+
+    private Mono<Void> sendRedirectResponse(HttpServerResponse response) {
+        return response.status(204)
```

Comment:
The legacy openpaas is return 302
with header Location = `/#/calendar`

Should we make it similar in tcalendar-side-service?

Reply:
302 + redirect to calendar.spa.orl in configuration.properties is IMO acceptable.